packages/http-server/src/server.ts

```diff
@@ -97,67 +110,86 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
         inputOutputValidationErrors.forEach(validation => {
           const message = `Violation: ${validation.location.join('.') || ''} ${validation.message}`;
           if (validation.severity === DiagnosticSeverity[DiagnosticSeverity.Error]) {
-            request.log.error({ name: 'VALIDATOR' }, message);
+            components.logger.error({ name: 'VALIDATOR' }, message);
           } else if (validation.severity === DiagnosticSeverity[DiagnosticSeverity.Warning]) {
-            request.log.warn({ name: 'VALIDATOR' }, message);
+            components.logger.warn({ name: 'VALIDATOR' }, message);
           } else {
-            request.log.info({ name: 'VALIDATOR' }, message);
+            components.logger.info({ name: 'VALIDATOR' }, message);
           }
         });
 
         return TE.fromIOEither(() =>
           E.tryCatch(() => {
-            if (output.headers) reply.headers(output.headers);
+            if (output.headers)
+              Object.entries(output.headers).forEach(([name, value]) => reply.setHeader(name, value));
 
-            reply
-              .code(output.statusCode)
-              .serializer((payload: unknown) => serialize(payload, reply.getHeader('content-type')))
-              .send(output.body);
+            send(
+              reply,
+              output.statusCode,
+              serialize(output.body, reply.getHeader('content-type') as string | undefined)
```

Comment:
Hmm — the cast is effectively legit. I guess there is nothing we can do about it here.